lib/bmp

```diff
@@ -15,16 +15,12 @@ u32le() {
 	# octet2: 0x34
 	# octet3: 0x56
 	# octet4: 0x78
-	local octet1=$(( (n >> 24) & 0xFF))
-	local octet2=$(( (n >> 16) & 0xFF))
-	local octet3=$(( (n >> 8)  & 0xFF))
-	local octet4=$(( (n >> 0)  & 0xFF))
-
-	printf -v out '\\x%02x\\x%02x\\x%02x\\x%02x' \
-	    "$octet4" \
-	    "$octet3" \
-	    "$octet2" \
-	    "$octet1"
+	local octet1=$(( (n >> 0)  & 0xFF))
+	local octet2=$(( (n >> 8)  & 0xFF))
+	local octet3=$(( (n >> 16) & 0xFF))
+	local octet4=$(( (n >> 24) & 0xFF))
+
+	printf -v out '\\x%02x' $octet1 $octet2 $octet3 $octet4
```

Comment:
i'm all for reducing the format string to just be 1, but the variables should be quoted (even though they are numbers and *shouldn't* contain any whitespace).